apis/v1alpha1/opentelemetrycollector_types.go

```diff
@@ -361,6 +361,11 @@ type OpenTelemetryTargetAllocator struct {
 	// +kubebuilder:validation:Optional
 	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Observability"
 	Observability ObservabilitySpec `json:"observability,omitempty"`
+	// CollectorNotReadyGracePeriod defines the grace period of a TargetAllocator stops considering a collector is target assignable.
+	// The default is 0s.
+	//
+	// +optional
+	CollectorNotReadyGracePeriod string `json:"collectorNotReadyGracePeriod,omitempty"`
```

Comment:
you should be able to use the `metav1.Duration` type here which supports marshalling.